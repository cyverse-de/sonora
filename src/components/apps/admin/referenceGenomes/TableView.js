/**
 * @author sriram
 *
 * A table view for admin Reference Genome Listing
 *
 */

import React, { useState } from "react";

import {
    useRowSelect,
    useGlobalFilter,
    useSortBy,
    useTable,
} from "react-table";

import refGenomeConstants from "./constants";
import TableToolbar from "./Toolbar";

import ids from "./ids";

import buildID from "components/utils/DebugIDUtil";

import Checkbox from "@mui/material/Checkbox";
import Table from "@mui/material/Table";
import Paper from "@mui/material/Paper";

import TableBody from "@mui/material/TableBody";
import TableCell from "@mui/material/TableCell";
import TableContainer from "@mui/material/TableContainer";
import TableHead from "@mui/material/TableHead";
import TableRow from "@mui/material/TableRow";
import TableSortLabel from "@mui/material/TableSortLabel";

const IndeterminateCheckbox = React.forwardRef(function IndeterminateCheckbox(
    { indeterminate, ...rest },
    ref
) {
    const defaultRef = React.useRef();
    const resolvedRef = ref || defaultRef;

    React.useEffect(() => {
        resolvedRef.current.indeterminate = indeterminate;
    }, [resolvedRef, indeterminate]);

    return <Checkbox type="Checkbox" ref={resolvedRef} {...rest} />;
});

const EnhancedTable = ({
    baseId,
    columns,
    data,
    onAddClicked,
    onDeletedClicked,
}) => {
    const [lastSelectIndex, setLastSelectIndex] = useState(-1);
    const {
        getTableProps,
        headerGroups,
        prepareRow,
        rows,
        preGlobalFilteredRows,
        setGlobalFilter,
        selectedFlatRows,
        state: { globalFilter },
        toggleRowSelected,
    } = useTable(
        {
            columns,
            data,
            disableSortBy: true,
            initialState: {
                sortBy: [
                    {
                        id: refGenomeConstants.keys.NAME,
                        desc: false,
                    },
                ],
            },
        },
        useGlobalFilter,
        useSortBy,
        useRowSelect,
        (hooks) => {
            hooks.allColumns.push((columns) => [
                // Let's make a column for selection
                {
                    id: "selection",
                    // The header can use the table's getToggleAllRowsSelectedProps method
                    // to render a checkbox
                    Header: ({ getToggleAllRowsSelectedProps }) => (
                        <IndeterminateCheckbox
                            {...getToggleAllRowsSelectedProps()}
                        />
                    ),
                    // The cell can use the individual row's getToggleRowSelectedProps method
                    // to the render a checkbox
                    Cell: ({ row }) => (
                        <IndeterminateCheckbox
                            {...row.getToggleRowSelectedProps()}
                        />
                    ),
                },
                ...columns,
            ]);
        }
    );

    const select = (rows) => {
        let newSelected = [...new Set([...selectedFlatRows, ...rows])];
        newSelected.forEach((row) => toggleRowSelected(row.index, true));
    };

    const deselect = (rows) => {
        const newSelected = selectedFlatRows.filter((row) =>
            rows.includes(row)
        );

        newSelected.forEach((row) => toggleRowSelected(row.index, false));
    };

    const rangeSelect = (start, end, row) => {
        if (start > -1) {
            const rangeIds = [];
            for (let i = start; i <= end; i++) {
                rangeIds.push(rows[i]);
            }

            let isTargetSelected = selectedFlatRows.includes(row);
            isTargetSelected ? deselect(rangeIds) : select(rangeIds);
        }
    };

    const handleClick = (event, row, index) => {
        if (event.shiftKey) {
            lastSelectIndex > index
                ? rangeSelect(index, lastSelectIndex, row)
                : rangeSelect(lastSelectIndex, index, row);
        } else {
            toggleRowSelected(index, !row.isSelected);
        }

        setLastSelectIndex(index);
    };

    const tableId = buildID(baseId, ids.TABLE_VIEW);
    return (
        <>
            <TableToolbar
                baseId={tableId}
                preGlobalFilteredRows={preGlobalFilteredRows}
                setGlobalFilter={setGlobalFilter}
                globalFilter={globalFilter}
                onAddClicked={onAddClicked}
                onDeletedClicked={onDeletedClicked}
            />
            <TableContainer component={Paper} style={{ overflow: "auto" }}>
                <Table size="small" stickyHeader {...getTableProps()}>
                    <TableHead>
                        {headerGroups.map((headerGroup) => (
                            <TableRow
                                key={headerGroup.id}
                                {...headerGroup.getHeaderGroupProps()}
                            >
                                {headerGroup.headers.map((column) => (
                                    <TableCell
                                        key={column.id}
                                        {...(column.id === "selection"
                                            ? column.getHeaderProps()
                                            : column.getHeaderProps(
                                                  column.getSortByToggleProps()
                                              ))}
                                    >
                                        {column.render("Header")}
                                        {column.id !== "selection" ? (
                                            <TableSortLabel
                                                active={column.isSorted}
                                                direction={
                                                    column.isSortedDesc
                                                        ? "desc"
                                                        : "asc"
                                                }
                                            />
                                        ) : null}
                                    </TableCell>
                                ))}
                            </TableRow>
                        ))}
                    </TableHead>
                    <TableBody>
                        {rows.map((row, index) => {
                            prepareRow(row);
                            return (
                                <TableRow
                                    key={row.id}
                                    {...row.getRowProps()}
                                    onClick={(event) =>
                                        handleClick(event, row, index)
                                    }
                                >
                                    {row.cells.map((cell) => {
                                        return (
                                            <TableCell
                                                key={cell.row.id}
                                                {...cell.getCellProps()}
                                            >
                                                {cell.render("Cell")}
                                            </TableCell>
                                        );
                                    })}
                                </TableRow>
                            );
                        })}
                    </TableBody>
                </Table>
            </TableContainer>
        </>
    );
};
export default EnhancedTable;
