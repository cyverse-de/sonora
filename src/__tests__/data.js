import React from "react";
import renderer from "react-test-renderer";

import { DataTableViewTest } from "../../stories/data/TableView.stories";

test("Data Table View renders", () => {
    const component = renderer.create(<DataTableViewTest />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
