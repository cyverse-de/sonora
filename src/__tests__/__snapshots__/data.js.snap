// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Table View renders 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <table
    aria-label={
      <FormattedMessage
        id="ariaTableListing"
        values={
          Object {
            "path": "/iplant/home/ipcdev",
          }
        }
      />
    }
    className="MuiTable-root MuiTable-stickyHeader"
    id="tableView.listingTable"
  >
    <thead
      className="MuiTableHead-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head EnhancedTableHead-checkbox_cell-33 MuiTableCell-paddingCheckbox MuiTableCell-stickyHeader"
          scope="col"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-58 MuiCheckbox-root MuiCheckbox-colorSecondary EnhancedTableHead-column_heading-32 MuiIconButton-colorSecondary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                checked={false}
                className="PrivateSwitchBase-input-61"
                data-indeterminate={false}
                onChange={[Function]}
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </span>
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head EnhancedTableHead-column_heading-32 MuiTableCell-alignCenter MuiTableCell-stickyHeader"
          id="tableView.listingTable."
          scope="col"
        >
          
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head EnhancedTableHead-column_heading-32 MuiTableCell-alignLeft MuiTableCell-stickyHeader"
          id="tableView.listingTable."
          scope="col"
        >
          <span
            aria-describedby={null}
            aria-disabled={false}
            className="MuiButtonBase-root MuiTableSortLabel-root"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            style={
              Object {
                "color": "#ffffff",
              }
            }
            tabIndex={0}
            title="Sort"
          >
            Name
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head EnhancedTableHead-column_heading-32 MuiTableCell-alignLeft MuiTableCell-stickyHeader"
          id="tableView.listingTable."
          scope="col"
        >
          Last Modified
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head EnhancedTableHead-column_heading-32 MuiTableCell-alignLeft MuiTableCell-stickyHeader"
          id="tableView.listingTable."
          scope="col"
        >
          Size
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head EnhancedTableHead-column_heading-32 MuiTableCell-alignLeft MuiTableCell-stickyHeader"
          id="tableView.listingTable."
          scope="col"
        >
          Path
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head EnhancedTableHead-column_heading-32 MuiTableCell-alignLeft MuiTableCell-stickyHeader"
          id="tableView.listingTable."
          scope="col"
        >
          
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
    >
      <tr
        aria-checked={false}
        className="MuiTableRow-root makeStyles-root-100 MuiTableRow-hover makeStyles-hover-102"
        id="tableView.listingTable.analyses_qa-3"
        onClick={[Function]}
        role="checkbox"
        tabIndex={0}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-58 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                aria-label="analyses_qa-3 checkbox"
                checked={false}
                className="PrivateSwitchBase-input-61"
                data-indeterminate={false}
                id="tableView.listingTable.analyses_qa-3.checkbox"
                onChange={[Function]}
                tabIndex={0}
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
            />
          </svg>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="makeStyles-dataLink-103"
            id="tableView.listingTable.analyses_qa-3.navigationLink"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="link"
            tabIndex={0}
          >
            analyses_qa-3
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          2016 Sep 20 20:27:57
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          -
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          /iplant/home/ipcdev/analyses_qa-3
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <button
            aria-controls="tableView.listingTable.analyses_qa-3.moreMenu"
            aria-haspopup="true"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
            disabled={false}
            id="tableView.listingTable.analyses_qa-3.dotMenuIcon"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </td>
      </tr>
      <tr
        aria-checked={false}
        className="MuiTableRow-root makeStyles-root-100 MuiTableRow-hover makeStyles-hover-102"
        id="tableView.listingTable.test"
        onClick={[Function]}
        role="checkbox"
        tabIndex={0}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-58 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                aria-label="test checkbox"
                checked={false}
                className="PrivateSwitchBase-input-61"
                data-indeterminate={false}
                id="tableView.listingTable.test.checkbox"
                onChange={[Function]}
                tabIndex={0}
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
            />
          </svg>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="makeStyles-dataLink-103"
            id="tableView.listingTable.test.navigationLink"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="link"
            tabIndex={0}
          >
            test
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          2018 Jun 28 22:49:35
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          -
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          /iplant/home/ipcdev/test
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <button
            aria-controls="tableView.listingTable.test.moreMenu"
            aria-haspopup="true"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
            disabled={false}
            id="tableView.listingTable.test.dotMenuIcon"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </td>
      </tr>
      <tr
        aria-checked={false}
        className="MuiTableRow-root makeStyles-root-100 MuiTableRow-hover makeStyles-hover-102"
        id="tableView.listingTable.CORE-8424.txt"
        onClick={[Function]}
        role="checkbox"
        tabIndex={0}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-58 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                aria-label="CORE-8424.txt checkbox"
                checked={false}
                className="PrivateSwitchBase-input-61"
                data-indeterminate={false}
                id="tableView.listingTable.CORE-8424.txt.checkbox"
                onChange={[Function]}
                tabIndex={0}
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
            />
          </svg>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="makeStyles-dataLink-103"
            id="tableView.listingTable.CORE-8424.txt.navigationLink"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="link"
            tabIndex={0}
          >
            CORE-8424.txt
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          2017 Feb 08 00:03:33
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          9 B
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          /iplant/home/ipcdev/CORE-8424.txt
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <button
            aria-controls="tableView.listingTable.CORE-8424.txt.moreMenu"
            aria-haspopup="true"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
            disabled={false}
            id="tableView.listingTable.CORE-8424.txt.dotMenuIcon"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </td>
      </tr>
      <tr
        aria-checked={false}
        className="MuiTableRow-root makeStyles-root-100 MuiTableRow-hover makeStyles-hover-102"
        id="tableView.listingTable.sample.csv"
        onClick={[Function]}
        role="checkbox"
        tabIndex={0}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-58 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                aria-label="sample.csv checkbox"
                checked={false}
                className="PrivateSwitchBase-input-61"
                data-indeterminate={false}
                id="tableView.listingTable.sample.csv.checkbox"
                onChange={[Function]}
                tabIndex={0}
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
            />
          </svg>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <span
            className="makeStyles-dataLink-103"
            id="tableView.listingTable.sample.csv.navigationLink"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="link"
            tabIndex={0}
          >
            sample.csv
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          2014 Oct 30 20:23:43
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          15 B
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          /iplant/home/ipcdev/sample.csv
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <button
            aria-controls="tableView.listingTable.sample.csv.moreMenu"
            aria-haspopup="true"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
            disabled={false}
            id="tableView.listingTable.sample.csv.dotMenuIcon"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
